src/rai_extensions/rai_open_set_vision/rai_open_set_vision/services/grounded_sam.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (C) 2024 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import subprocess
+from pathlib import Path
+
+import rclpy
+from rai_open_set_vision.vision_markup.segmenter import GDSegmenter
+from rclpy.node import Node
+
+from rai_interfaces.srv import RAIGroundedSam
+
+GSAM_NODE_NAME = "grounded_sam"
+GSAM_SERVICE_NAME = "grounded_sam_segment"
+
+
+# TODO: Create a base class for vision services
+class GSamService(Node):
+    def __init__(self):
+        super().__init__(node_name=GSAM_NODE_NAME, parameter_overrides=[])
+        self.srv = self.create_service(
+            RAIGroundedSam, GSAM_SERVICE_NAME, self.segment_callback
+        )
+
+        self.declare_parameter("weights_path", "")
+        try:
+            weight_path = self.get_parameter("weights_path").value
+            assert isinstance(weight_path, str)
+            if self.get_parameter("weights_path").value == "":
+                weight_path = self._init_weight_path()
+            self.segmenter = GDSegmenter(weight_path)
+        except Exception:
+            self.get_logger().error("Could not load model")
+            raise Exception("Could not load model")
+
+    def _init_weight_path(self):
+        try:
+            found_path = subprocess.check_output(
+                ["ros2", "pkg", "prefix", "rai_open_set_vision"]
+            ).decode("utf-8")
+            install_path = (
+                Path(found_path.strip()) / "share" / "weights" / "sam2_hiera_large.pt"
+            )
+            # make sure the file exists
+            if install_path.exists():
+                return install_path
+            else:
+                self._download_weights(install_path)
```

Comment:
Can you please reduce the verbosity of the downloading? The first time this script is ran it absolutely destroys my terminal :D

Reply:
This is actually a big tradeoff - given that the download is performed by the `wget` subprocess, we are limited to the `-q` option which completely supresses _all_ output from wget. This includes error and warning output as well - so in  case of an error this would leave the user clueless as to what happened.

It is possible to handle errors through checking the error codes of the process, in accordance to the following (from `man wget`:

```
      Wget may return one of several error codes if it encounters problems.

       0   No problems occurred.
       1   Generic error code.
       2   Parse error---for instance, when parsing command-line options, the .wgetrc or .netrc...
       3   File I/O error.
       4   Network failure.
       5   SSL verification failure.
       6   Username/password authentication failure.
       7   Protocol errors.
       8   Server issued an error response.

       With the exceptions of 0 and 1, the lower-numbered exit codes take precedence over higher-numbered ones, when multiple types of errors are encountered.

       In versions of Wget prior to 1.12, Wget's exit status tended to be unhelpful and inconsistent. Recursive downloads would virtually always return 0 (success),
       regardless of any issues encountered, and non-recursive fetches only returned the status corresponding to the most recently-attempted download.
```

This will give some verbosity regarding errors, while also allowing for supressing the output -- but... I am not sure if this is the most robust way. Unless we decide to start considering `wget` download an idiomatic part of RAI processes, and include the download+error handling in `rai.core`, which would allow to make this error handling consistent across RAI packages.

Reply:
For now let's keep it as it is. Could you please add an issue regarding this?